science/imager/Portfile

```
@@ -0,0 +1,101 @@
+
+set my_build_opts   "-o openmp"
+#variant openmp description {Add OpenMP support} {set my_build_opts "-o openmp"}
+
+# need x11 and cannot be used with quartz; see ticket #42886
+require_active_variants gtk2 x11 quartz
+
+# need the fortran interface; see ticket #50543
+compilers.enforce_some_fortran cfitsio
+
+compilers.choose    fc
+compilers.setup     -gcc46 require_fortran
+configure.python    ${prefix}/bin/python3.10
+
+configure {
+    set cxxstdlib {}
+    if {[string match *clang* ${configure.cxx}]} {
+        set cxxstdlib [string map {lib -l} ${configure.cxx_stdlib}]
+    }
+
+    reinplace -W ${worksrcpath}/admin "s|@PREFIX@|${prefix}|g" define-system.sh gildas-env.sh
+    reinplace -W ${worksrcpath}/admin "s|@CC@|${configure.cc}|g" define-system.sh
+    reinplace -W ${worksrcpath}/admin "s|@CCFLAGS@|${configure.cflags}|g" Makefile.def
+    reinplace -W ${worksrcpath}/admin "s|@FC@|${configure.fc}|g" define-system.sh
+    reinplace -W ${worksrcpath}/admin "s|@FCFLAGS@|${configure.fcflags}|g" Makefile.def
+    reinplace -W ${worksrcpath}/admin "s|@CXX@|${configure.cxx}|g" define-system.sh
+    reinplace -W ${worksrcpath}/admin "s|@CPP@|${configure.cpp}|g" Makefile.def define-system.sh
+    reinplace -W ${worksrcpath}/admin "s|@CPPFLAGS@|${configure.cppflags}|g" Makefile.def
+    reinplace -W ${worksrcpath}/admin "s|@LDFLAGS@|${configure.ldflags}|g" Makefile.def
+    reinplace -W ${worksrcpath}/admin "s|@PYTHON@|${configure.python}|g" Makefile.build Makefile.def Makefile.python gildas-env.sh
+}
+
+build {
+    system -W ${worksrcpath} "source admin/gildas-env.sh -c ${configure.fc} ${my_build_opts} -s ${prefix}/include:${prefix}/lib:/System/Library/Frameworks/Accelerate.framework/Frameworks/vecLib.framework/Versions/Current/ && export GAG_SLDFLAGS='-shared -o ${prefix}/lib/gildas/${my_machine}-darwin-gfortran/lib/\$(notdir \$@) -install_name ${prefix}/lib/gildas/${my_machine}-darwin-gfortran/lib/\$(notdir \$@)' && export DYLD_LIBRARY_PATH=${worksrcpath}/integ/${my_machine}-darwin-gfortran/lib && export GAG_ADDONS=yes && make -w install"
+}
+
+# Attention: gildas-exe ou imager-exe ?...
+destroot {
+    copy ${workpath}/gildas-exe-${my_version} ${destroot}${prefix}/lib/imager
+
+    xinstall ${filespath}/greg.in ${destroot}${prefix}/bin/greg
+    xinstall ${filespath}/sic.in ${destroot}${prefix}/bin/sic
+    xinstall ${filespath}/imager.in ${destroot}${prefix}/bin/imager
+    xinstall ${filespath}/viewer.in ${destroot}${prefix}/bin/viewer
+
+#    reinplace -W ${destroot}${prefix}/bin s|@PREFIX@|${prefix}|g astro class clic cube greg mapping mira mrtcal sic sched-30m imager
+#    reinplace -W ${destroot}${prefix}/bin s|@ARCH@|${my_machine}|g astro class clic cube greg mapping mira mrtcal sic sched-30m imager
+    reinplace -W ${destroot}${prefix}/bin s|@PREFIX@|${prefix}|g greg sic imager viewer
+    reinplace -W ${destroot}${prefix}/bin s|@ARCH@|${my_machine}|g greg sic imager viewer
+
+    
```

This Portfile doesn't parse because you're missing a closing brace:
```
% sudo port build
Can't map the URL 'file://.' to a port description file ("missing close-brace: possible unbalanced brace in comment").
Please verify that the directory and portfile syntax are correct.
To use the current port, you must be in a port's directory.
```
```suggestion
}
```

---

correction done